src/scripts/main.js

```diff
@@ -1,3 +1,45 @@
 'use strict';
 
-// write code here
+const list = document.querySelectorAll('ul li');
+
+function sortList() {
+  const listItems = [...list];
+
+  const numeric = () => {
+    for (const item of listItems) {
+      item.dataset.salary = Number(
+        item.dataset.salary.replace(/[^0-9.-]+/g, ''),
+      );
+    }
+  };
+
+  numeric();
+
+  listItems.sort((a, b) => b.dataset.salary - a.dataset.salary);
+
+  const ul = document.querySelector('ul');
+
+  listItems.forEach((item) => {
+    ul.appendChild(item);
+  });
+}
+
+function getEmployees() {
+  const arrayEmployees = [];
+
+  list.forEach((item) => {
+    const employee = {
+      name: item.textContent,
+      position: item.dataset.position,
+      salary: item.dataset.salary,
```

Comment:
Similar to the salary attribute, ensure that 'data-position', 'data-salary', and 'data-age' attributes exist on each 'li' element to avoid undefined values in the employee object.